### PR TITLE
PLATT-43: nearby widget to PDP

### PR DESCRIPTION
**Pull request by cole-sanderson**
Add Nearby Widget to PDP

 **JIRA**: https://mobify.atlassian.net/browse/PLATT-43
 **Linked PRs**: https://github.com/mobify/platform-scaffold/pull/349

## Changes
- add Nearby Widget to PDP

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Go to one of PDP
- Make sure that nearby widget script only being added on PDP, it should not be added on any other pages than PDP.

### CODE REVIEW

File: web/app/containers/product-details/partials/product-nearest-stores.jsx [hunk 1]
```diff
@@ -1,23 +1,29 @@
...
 import Button from 'progressive-web-sdk/dist/components/button'
 import ListTile from 'progressive-web-sdk/dist/components/list-tile'
 import SkeletonText from 'progressive-web-sdk/dist/components/skeleton-text'
 import SkeletonBlock from 'progressive-web-sdk/dist/components/skeleton-block'
+import Icon from 'progressive-web-sdk/dist/components/icon'
 
 // Merlins Potions Nearby Widget Config
 import merlinsPotionsNearbyConfig from '../../../config/merlins-potions-nearby-config.json'
 
+// Merlins Potions Nearby Widget ID selector
```
Comment (jeremywiebe) [inline web/app/containers/product-details/partials/product-nearest-stores.jsx]: I think this comment and the one above are self-evident and redundant based on the variable names.

File: web/app/containers/product-details/partials/product-nearest-stores.jsx [hunk 2]
```diff
@@ -1,23 +1,29 @@
...
+class ProductNearestStores extends React.Component {
 
     componentWillMount() {
         // Nearby widget async script
         const merlinsPotionsAsync = !function(a) { // eslint-disable-line wrap-iife
             const b = document.createElement('script')
+            b.setAttribute('id', $merlinsPotionsNearbyWidgetSelector)
             b.type = 'text/javascript'
             b.src = 'https://dtopnrgu570sp.cloudfront.net/nearby-widget/nearby.min.js'
             b.setAttribute('async', !0)
```
Comment (jeremywiebe) [inline web/app/containers/product-details/partials/product-nearest-stores.jsx]: I couldn't add a comment on the next line, but the formatting of that `addEventListener` stuff is a bit hard to follow. One pattern i've found really nice is formatting ternary's across 3 lines with the `?` and `:` at the start of line 2 and 3... like this:

```js
b.addEventListener 
    ? b.addEventListener('load', (b) => { // eslint-disable-line no-unused-expressions
        a(null, b)
    }, !1) 
    : b.onreadystatechange = function() {
        b.readyState in { // eslint-disable-line no-unused-expressions
            loaded: 1,
            complete: 1
        } && (b.onreadystatechange = null, a())
    }
```

I'm also confused over the use of `!1` for the `useCapturing` option.  Why not simply `false`?

File: web/app/containers/product-details/partials/product-nearest-stores.jsx [hunk 3]
```diff
@@ -39,6 +45,12 @@ class HomeNearestStores extends React.Component {
         document.body.appendChild(merlinsPotionsScript)
     }
 
+    componentWillUnmount() {
+        // Remove nearby widget script if component unmount
+        const $merlinsPotionsNearbyWidgetScript = document.getElementById($merlinsPotionsNearbyWidgetSelector)
```
Comment (jeremywiebe) [inline web/app/containers/product-details/partials/product-nearest-stores.jsx]: Given that this `$merlinsPotionsNearbyWidgetScript` variable is only used on this line and the next, it's often easier to understand what's going on by shortening the name (a general rule is that the smaller the scope, the shorter the variable name).

```js
const $script = document.getElementById($merlinsPotionsNearbyWidgetSelector)
$script.parentNode.removeChild($script)
```

File: web/app/containers/product-details/partials/product-nearest-stores.jsx (hunk 1 above)
Comment (cole-sanderson) [inline web/app/containers/product-details/partials/product-nearest-stores.jsx]: 👍

File: web/app/containers/product-details/partials/product-nearest-stores.jsx (hunk 2 above)
Comment (cole-sanderson) [inline web/app/containers/product-details/partials/product-nearest-stores.jsx]: 👍

File: web/app/containers/product-details/partials/product-nearest-stores.jsx (hunk 3 above)
Comment (cole-sanderson) [inline web/app/containers/product-details/partials/product-nearest-stores.jsx]: 👍

**Comment by cole-sanderson** [comment]
@jeremywiebe Thank you for leaving some comments. I just pushed the change.

File: web/app/containers/product-details/partials/product-nearest-stores.jsx (hunk 2 above)
Comment (cole-sanderson) [inline web/app/containers/product-details/partials/product-nearest-stores.jsx]: Getting "Expected an assignment or function call and instead aw an expression" error with your suggestion. Wont be able to do that.

Comment (jeremywiebe) [inline web/app/containers/product-details/partials/product-nearest-stores.jsx]: Yeah, I think it's because it doesn't understand that `b.addEventListener` on the first line is a boolean check (which it becomes if you follow it with a `?`).  I'm ok to leave it as it is for now.
